exp-player/addon/components/exp-rating-form/component.js

```diff
@@ -4,569 +4,580 @@ import layout from './template';
+        'measures.questions.15.items.8.label',
+        'measures.questions.15.items.9.label',
+        'measures.questions.15.items.10.label'
+    ],
+    '16': [
+        'measures.questions.16.items.1.label',
+        'measures.questions.16.items.2.label',
+        'measures.questions.16.items.3.label',
+        'measures.questions.16.items.4.label',
+        'measures.questions.16.items.5.label'
+    ],
+    '17': [
+        'measures.questions.17.items.1.label',
+        'measures.questions.17.items.2.label',
+        'measures.questions.17.items.3.label',
+        'measures.questions.17.items.4.label',
+        'measures.questions.17.items.5.label',
+        'measures.questions.17.items.6.label'
+    ]
 };
 
 const TEN_POINT_SCALE = [
-  'number0',
-  'number1',
-  'number2',
-  'number3',
-  'number4',
-  'number5',
-  'number6',
-  'number7',
-  'number8',
-  'number9',
-  'number10'
+    'number0',
+    'number1',
+    'number2',
+    'number3',
+    'number4',
+    'number5',
+    'number6',
+    'number7',
+    'number8',
+    'number9',
+    'number10'
 ];
 const SEVEN_POINT_SCALE = TEN_POINT_SCALE.slice(0, 8);
 const NINE_POINT_SCALE = TEN_POINT_SCALE.slice(0, 10);
 
-var generateValidators = function(questions) {
-  var validators = {};
-  var presence = validator('presence', {
-      presence:true,
-      message: 'This field is required'
-  });
-  for (var i=0; i < questions.length; i++) {
-    for (var item=0; item < questions[i]['items'].length; item++) {
-      var isOptional = questions[i]['items'][item].optional;
-      if (!isOptional) {
-        var key = `questions.${i}.items.${item}.value`;
-        validators[key] = presence;
-      }
+var generateValidators = function (questions) {
+    var validators = {};
+    var presence = validator('presence', {
+        presence: true,
+        message: 'This field is required'
+    });
+    for (var i = 0; i < questions.length; i++) {
+        for (var item = 0; item < questions[i]['items'].length; item++) {
+            var isOptional = questions[i]['items'][item].optional;
+            if (!isOptional) {
+                var key = `questions.${i}.items.${item}.value`;
+                validators[key] = presence;
+            }
+        }
     }
-  }
-  return validators;
+    return validators;
 };
 
-var generateSchema = function(question, type, items, scale, options) {
-  var schema = {
-    question: question,
-    type: type,
-    scale: scale,
-    items: []
-  };
-  for (var i=0; i < items.length; i++) {
-    var ret = {
-      description: items[i],
-      value: null
-    };
-    for (var option in options) {
-      ret[option] = options[option];
+var generateSchema = function (data) {
+    var items = [];
+    var options = data.options ? data.options : {};
+    for (var i = 0; i < data.items.length; i++) {
+        var ret = {
+            description: data.items[i],
+            value: null
+        };
+        for (var option in options) {
```

Comment:
What about `Object.keys(options).forEach(...)`

The `hasOwnProperty` check seems less than ideal.